org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/PersistentModel.java

```diff
@@ -9,76 +9,41 @@
 package org.eclipse.buildship.core.preferences;
 
 import java.util.Collection;
+import java.util.List;
 
 import com.google.common.base.Optional;
 
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IResource;
 import org.eclipse.core.runtime.IPath;
+import org.eclipse.jdt.core.IClasspathEntry;
 
 /**
- * Interface to load and store Gradle model elements stored in the workspace plugin state area. The
- * model elements can be stored and retrieved in either a structured or in a map-like fashion.
+ * Interface to load and store Gradle model elements stored in the workspace plugin state area.
+ * <p/>
+ * The {@link Optional} return type on the getters mark whether a value is present in the model.
  *
  * @author Donat Csikos
  */
 public interface PersistentModel {
 
-    /**
-     * Returns the build directory.
-     * @return the build directory
-     */
     Optional<IPath> getBuildDir();
```

Comment:
Why is this an IPath instead of an IResource? Then we could just do a simple equals where it is queried.